data-collection/src/main/java/refactoringml/RunQueue.java

```diff
@@ -72,6 +73,8 @@ private void run() throws IOException, TimeoutException {
 					doWork(message);
 				}
 			}
+			//force garbage collection to free some memory
+			System.gc();
```

Comment:
It is usually not a good idea to call `System.gc()` manually. The GC is smart enough to know when to clean up memory.

Often, manually deciding GC points actually harm the performance (because you end up calling the GC way more than needed). I'd say that, without some fine-grained performance analysis that shows the need for a gc call, this line is not a good idea.

Reply:
Ok, I will remove it